### PR TITLE
Implementing Pagination

### PR DESCRIPTION
**Pull request by MariaCrucin**

### CODE REVIEW

**Review by cataraul — APPROVED** [review]

File: TweetBook/Controllers/V1/PostsController.cs [hunk 1]
```diff
@@ -18,20 +20,29 @@ public class PostsController : ControllerBase
...
         private readonly IMapper _mapper;
+        private readonly IUriService _uriService;
 
-        public PostsController(IPostService postService, IMapper mapper)
+        public PostsController(IPostService postService, IMapper mapper, IUriService uriService)
         {
             _postService = postService;
             _mapper = mapper;
+            _uriService = uriService;
         }
```
Comment (SebiBarson) [inline TweetBook/Controllers/V1/PostsController.cs:31]: You could document this endpoint

File: TweetBook/Controllers/V1/PostsController.cs [hunk 2]
```diff
@@ -18,20 +20,29 @@ public class PostsController : ControllerBase
...
+
+            var postsResponse = _mapper.Map<List<PostResponse>>(posts);
 
-            return Ok(_mapper.Map<List<PostResponse>>(posts));
+            if (pagination == null || pagination.PageNumber < 1 || pagination.PageSize < 1)
+                return Ok(new PagedResponse<PostResponse>(postsResponse));
+
+            var paginationResponse = PaginationHelpers.CreatePaginatedResponse(_uriService, pagination, postsResponse);
+            return Ok(paginationResponse);
         }
```
Comment (SebiBarson) [inline TweetBook/Controllers/V1/PostsController.cs:53]: This one too

File: TweetBook/Controllers/V1/PostsController.cs [hunk 3]
```diff
@@ -42,7 +53,7 @@ public async Task<IActionResult> Get([FromRoute] Guid postId)
             if (post == null)
                 return NotFound();
 
-            return Ok(_mapper.Map<PostResponse>(post));
+            return Ok(new Response<PostResponse>(_mapper.Map<PostResponse>(post)));
         }
```
Comment (SebiBarson) [inline TweetBook/Controllers/V1/PostsController.cs:72]: Same for this one

File: TweetBook/Controllers/V1/PostsController.cs [hunk 4]
```diff
@@ -61,10 +72,9 @@ public async Task<IActionResult> Create([FromBody] CreatePostRequest postRequest
...
                 return BadRequest(new { error = "Unable to create post" });
 
-            var baseUrl = $"{HttpContext.Request.Scheme}://{HttpContext.Request.Host.ToUriComponent()}";
-            var locationUrl = $"{baseUrl}/{ApiRoutes.Posts.Get.Replace("{postId}", post.Id.ToString())}";
-
-            return Created(locationUrl, _mapper.Map<PostResponse>(post));
+            var locationUri = _uriService.GetPostUri(post.Id.ToString());
+                 
+            return Created(locationUri, new Response<PostResponse>(_mapper.Map<PostResponse>(post)));
         }
```
Comment (SebiBarson) [inline TweetBook/Controllers/V1/PostsController.cs:101]: Another one here

**Review by AlinCuc — APPROVED** [review]

File: TweetBook/Controllers/V1/PostsController.cs (hunk 1 above)
Comment (SebiBarson) [reply TweetBook/Controllers/V1/PostsController.cs:31]: Solved

File: TweetBook/Controllers/V1/PostsController.cs (hunk 2 above)
Comment (SebiBarson) [reply TweetBook/Controllers/V1/PostsController.cs:53]: Solved

File: TweetBook/Controllers/V1/PostsController.cs (hunk 3 above)
Comment (SebiBarson) [reply TweetBook/Controllers/V1/PostsController.cs:72]: Solved

File: TweetBook/Controllers/V1/PostsController.cs (hunk 4 above)
Comment (SebiBarson) [reply TweetBook/Controllers/V1/PostsController.cs:101]: Solved

**Review by SebiBarson — APPROVED** [review]
Good job :)

**Pull request merged**